tests/test_npy_cache.py

```diff
@@ -0,0 +1,112 @@
+    config_dict = yaml.safe_load(config_yaml)
+    config = VideoLoaderConfig(**config_dict)
+
+    # NOTE: the validation in VideoLoaderConfig changes some fields, so this fails
+    # assert dict(config) == config_dict
+
+    cached_load_video_frames = npy_cache(
+        cache_path=config.cache_dir, cleanup=config.cleanup_cache
+    )(load_video_frames)
+    assert isinstance(cached_load_video_frames, type(load_video_frames))
+
+    vid_path_str = "s3://drivendata-client-zamba/data/raw/noemie/Taï_cam197_683044_652175_20161223/01090065.AVI"
+    vid_path_s3 = S3Path(vid_path_str)
+    vid_path_path = Path(vid_path_s3.key)
+
+    expected = Path(
+        "data/cache/2d1fee2b1e1f78d06aa08bdea88e7661f927bd81/data/raw/noemie/Taï_cam197_683044_652175_20161223/01090065.npy"
+    )
+
+    # pass the vid_path as a string
+    path = get_cached_array_path(config, vid_path_str, config.cache_dir)
+    assert path == expected
+
+    # pass the vid_path as S3Path
+    path = get_cached_array_path(config, vid_path_s3, config.cache_dir)
+    assert path == expected
+
+    # pass the vid_path as a Path
+    path = get_cached_array_path(config, vid_path_path, config.cache_dir)
+    assert path == expected
+
+    # pass the cache_dir as a Path
+    path = get_cached_array_path(config, vid_path_path, Path(config.cache_dir))
+    assert path == expected
+
+    # pass the cache_dir as S3Path
+    cache_dir = S3Path("s3://drivendata-client-zamba") / config.cache_dir
+    path = get_cached_array_path(config, vid_path_path, cache_dir)
+    expected3 = S3Path(
+        "s3://drivendata-client-zamba/data/cache/2d1fee2b1e1f78d06aa08bdea88e7661f927bd81/data/raw/noemie/Taï_cam197_683044_652175_20161223/01090065.npy"
+    )
+    assert path == expected3
+
+    # pass the config as a dictionary
+    config_dict = config.dict()
+    path = get_cached_array_path(config_dict, vid_path_str, config.cache_dir)
+    assert path == expected
+
+    # pass the config as a dictionary, but use dict(config) rather than config.dict()
+    config_dict = dict(config)
+    path = get_cached_array_path(config_dict, vid_path_str, config.cache_dir)
+    # This fails because the string representation of the dictionary is different.
+    # But we don't need/want to test that it fails.
+    # assert path == expected
+
+    # changing config.cache_dir and/or config.cleanup_cache should not affect the key
+    config_dict = config.dict()
+    config_dict["cache_dir"] = "something/else"
+    config_dict["cleanup_cache"] = "not even bool"
+    path = get_cached_array_path(config_dict, vid_path_path, config.cache_dir)
+    assert path == expected
+
+    # changing anything else should
+    config_dict["total_frames"] = 8
+    path = get_cached_array_path(config_dict, vid_path_path, config.cache_dir)
+    expected2 = Path(
+        "data/cache/9becb6d6dfe6b9970afe05af06ef49af4881bd73/data/raw/noemie/Taï_cam197_683044_652175_20161223/01090065.npy"
+    )
+    assert path == expected2
```

Comment:
may be a touch clearer with

```
path_with_changed_total_frames = get_cached_array_path(config_dict, vid_path_path, config.cache_dir)
assert path_with_changed_total_frames != path
```

since the goal is to highlight that the path has changed, rather than that it matches `expected2`.

Reply:
Ok, but I also want to check the hash of a second case.